src/main/java/org/vietabroader/controller/ConnectController.java

```diff
@@ -0,0 +1,56 @@
+package org.vietabroader.controller;
+
+import org.vietabroader.GoogleAPIUtils;
+import org.vietabroader.model.GlobalState;
+import org.vietabroader.model.VASpreadsheet;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConnectController implements Controller {
+
+    private JButton btnConnect;
+    private JTextField txtConnect;
+    private static final Logger logger = LoggerFactory.getLogger(ConnectController.class);
+
+    public ConnectController setConnectButton(JButton btn) {
+        btnConnect = btn;
+        return this;
+    }
+
+    public ConnectController setTextSpreadsheetID(JTextField txt) {
+        txtConnect = txt;
+        return this;
+    }
+
+    @Override
+    public void control() {
+        btnConnect.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                GlobalState currentState = GlobalState.getInstance();
+                if (currentState.getStatus() == GlobalState.Status.SIGNED_OUT) {
```

Comment:
We can only go to `CONNECTED` when we `SIGNED_IN` so we don't care about this status here. Just remove this so that it does nothing.